packages/core/contracts/common/implementation/MultiCaller.sol

```diff
@@ -1,11 +1,13 @@
 // This contract is taken from Uniswaps's multi call implementation (https://github.com/Uniswap/uniswap-v3-periphery/blob/main/contracts/base/Multicall.sol)
-// and was modified to be solidity 0.8 compatible. No other changes were made.
+// and was modified to be solidity 0.8 compatible. Additionally, the method was restricted to only work with msg.value
+// set to 0 to avoid any nasty attack vectors on function calls that use value sent with deposits.
 pragma solidity ^0.8.0;
 
 /// @title MultiCaller
 /// @notice Enables calling multiple methods in a single call to the contract
 contract MultiCaller {
     function multicall(bytes[] calldata data) external payable returns (bytes[] memory results) {
+        require(msg.value == 0, "Only multicall with 0 value");
```

Comment:
Does this break functionality of any existing contracts that inherit MutliCaller?

Reply:
under all conditions multicall should not use any `msg.value`! this is where the dragon is!

also...we don't use this lib anywhere else in the repo :)